### ISSUE TITLE
XGBoost version not compiled with GPU support

### ISSUE THREAD

**Issue by chitransh89** [autogluon/autogluon#3957]
I am training XGBoost model using Autogluon 1.0.0. I trained using both GPU and CPU. However, when I am loading the trained model for inference and calling the predict method I am getting this error, `XGBoost version not compiled with GPU support`. The training server is Linux and inference is tested on MacOS.

Following is the training method call,
`
TabularPredictor(
                label=label,
                eval_metric="balanced_accuracy",
                learner_kwargs={
                    "ignored_columns": NONFEATURE_COLS,
                },
                sample_weight='sample_weights'
            ).fit(train_data,
                  tuning_data=val_data,
                  time_limit=12000,
                  hyperparameters=MODEL_PARAMETERS,
                    fit_weighted_ensemble=False,
                    num_stack_levels = 0,
                    ag_args_fit={ 'max_memory_usage_ratio': 0.9},
                    refit_full=True,
                    set_best_to_refit_full=True,
                    save_space=True,
                    excluded_model_types=['GBM','XGB','XT','RF'],
                    feature_prune_kwargs={'force_prune' : True}
                   )
`
I'd seen that it failed on doing refit_full but still it persisted the model on the disk and generated validation scores, hence I went ahead and put the model up for testing. 

This is the code for calling the inference,
**df pandas DataFrame has the input features**
```
predictor = TabularPredictor.load("models/AutogluonModels")
predictor.persist_models(models='best', max_memory=0.6, with_ancestors=False)
predictor.predict(df.iloc[[0]], as_pandas=False)[0]
```

<img width="1599" alt="Screenshot 2024-03-01 at 6 03 13 PM" src="https://github.com/autogluon/autogluon/assets/16305001/6a0e5844-5d4f-4e68-a72e-62388391fa1f">

**Installed Versions**
<details>

```python
INSTALLED VERSIONS
------------------
date                   : 2024-03-01
time                   : 18:05:46.218469
python                 : 3.10.12.final.0
OS                     : Darwin
OS-release             : 21.6.0
Version                : Darwin Kernel Version 21.6.0: Thu Jun  8 23:57:12 PDT 2023; root:xnu-8020.240.18.701.6~1/RELEASE_X86_64
machine                : x86_64
processor              : i386
num_cores              : 8
cpu_ram_mb             : 16384.0
cuda version           : None
num_gpus               : 0
gpu_ram_mb             : []
avail_disk_size_mb     : 373127

accelerate             : 0.21.0
async-timeout          : 4.0.3
autogluon              : 1.0.0
autogluon.common       : 1.0.0
autogluon.core         : 1.0.0
autogluon.features     : 1.0.0
autogluon.multimodal   : 1.0.0
autogluon.tabular      : 1.0.0
autogluon.timeseries   : 1.0.0
boto3                  : 1.29.3
catboost               : 1.1.1
defusedxml             : 0.7.1
evaluate               : 0.4.1
fastai                 : 2.7.13
gluonts                : 0.14.4
hyperopt               : 0.2.7
imodels                : None
jinja2                 : 3.1.2
joblib                 : 1.3.2
jsonschema             : 4.17.3
lightgbm               : 3.3.5
lightning              : 2.0.9.post0
matplotlib             : 3.8.2
mlforecast             : 0.10.0
networkx               : 3.2.1
nlpaug                 : 1.1.11
nltk                   : 3.8.1
nptyping               : 2.4.1
numpy                  : 1.26.2
nvidia-ml-py3          : 7.352.0
omegaconf              : 2.2.3
onnxruntime-gpu        : None
openmim                : 0.3.9
orjson                 : 3.9.15
pandas                 : 2.1.4
Pillow                 : 10.2.0
psutil                 : 5.9.6
PyMuPDF                : None
pytesseract            : 0.3.10
pytorch-lightning      : 2.0.9.post0
pytorch-metric-learning: 1.7.3
ray                    : 2.6.3
requests               : 2.28.2
scikit-image           : 0.19.3
scikit-learn           : 1.4.1.post1
scikit-learn-intelex   : None
scipy                  : 1.11.4
seqeval                : 1.2.2
setuptools             : 60.2.0
skl2onnx               : None
statsforecast          : 1.4.0
statsmodels            : 0.14.0
tabpfn                 : None
tensorboard            : 2.15.1
text-unidecode         : 1.3
timm                   : 0.9.10
torch                  : 2.0.1
torchmetrics           : 1.1.2
torchvision            : 0.15.2
tqdm                   : 4.65.2
transformers           : 4.31.0
utilsforecast          : 0.0.10
vowpalwabbit           : None
xgboost                : 1.7.6
```

</details>
